### ISSUE TITLE
False positive warning for the rule "prefer-destructuring"

### ISSUE THREAD

**Issue by mqklin** [eslint/eslint#13805]
**What did you do?**

```js
let foo;
let obj = {foo: 'bar'};
if (true) {
  foo = obj.foo; // error Use object destructuring  prefer-destructuring
}
```

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by danvk** [comment]
The rule wants you to write it this way:

```js
({foo} = obj);
```

**Comment by yeonjuan** [comment]
Hi 😁 @mqklin 

If you don't want to force destructuring in assignment expression, you can use the `AssignmentExpression` options. see [prefer-destructuring options](https://eslint.org/docs/rules/prefer-destructuring#options).
If it's not what you look for, would you mind fill the Bug Issue template or give us more description?.

* [online-demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwicHJlZmVyLWRlc3RydWN0dXJpbmdcIjogW1wiZXJyb3JcIiwge1xuXHRWYXJpYWJsZURlY2xhcmF0b3I6IHtcIm9iamVjdFwiOiB0cnVlfSxcblx0QXNzaWdubWVudEV4cHJlc3Npb246IHtcIm9iamVjdFwiOmZhbHNlfVxuICAgfVxuIF0qL1xuXG5sZXQgZm9vID0gYmFyLmZvbzsgLy8gRXJyb3IsIFVzZSBvYmplY3QgZGVzdHJ1Y3R1cmluZyBcbmxldCBvYmogPSB7Zm9vOiAnYmFyJ307XG5pZiAodHJ1ZSkge1xuICBmb28gPSBvYmouZm9vOyAvLyBObyBlcnJvclxufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiYnJvd3NlciI6dHJ1ZX19fQ==)

```js
/* eslint "prefer-destructuring": ["error", {
	VariableDeclarator: {"object": true},
	AssignmentExpression: {"object":false}
   }
 ]*/

let foo = bar.foo; // Error, Use object destructuring 
let obj = {foo: 'bar'};
if (true) {
  foo = obj.foo; // No error
}
```

**Comment by mqklin** [comment]
Thanks guys... Didn't know `({foo} = obj);` is possible 😲
